nodebuilder/tests/blob_test.go

```diff
@@ -123,12 +129,87 @@ func TestBlobModule(t *testing.T) {
 				require.ErrorContains(t, err, blob.ErrBlobNotFound.Error())
 			},
 		},
+		{
+			name: "Submit equal blobs",
+			doFn: func(t *testing.T) {
+				appBlob, err := blobtest.GenerateV0Blobs([]int{8, 4}, true)
+				require.NoError(t, err)
+				b, err := blob.NewBlob(
+					appBlob[0].ShareVersion,
+					append([]byte{appBlob[0].NamespaceVersion}, appBlob[0].NamespaceID...),
+					appBlob[0].Data,
+				)
+				require.NoError(t, err)
+
+				height, err := fullClient.Blob.Submit(ctx, []*blob.Blob{b, b}, nil)
+				require.NoError(t, err)
+
+				_, err = fullClient.Header.WaitForHeight(ctx, height)
+				require.NoError(t, err)
+
+				b0, err := fullClient.Blob.Get(ctx, height, b.Namespace(), b.Commitment)
+				require.NoError(t, err)
+				require.Equal(t, b, b0)
+
+				// give some time to store the data,
+				// otherwise the test will hang on the IPLD level.
+				// https://github.com/celestiaorg/celestia-node/issues/2915
+				time.Sleep(time.Second)
+
+				proof, err := fullClient.Blob.GetProof(ctx, height, b.Namespace(), b.Commitment)
+				require.NoError(t, err)
+
+				included, err := fullClient.Blob.Included(ctx, height, b.Namespace(), proof, b.Commitment)
+				require.NoError(t, err)
+				require.True(t, included)
+			},
+		},
+		{
+			name: "Submit equal blobs in different PFBs",
```

Comment:
what's the point of this test case if they're not in the same block?

Reply:
> Also, let's add a case where we see that the namespace does have two equal blobs. It might deduplicate the same blobs on in a single PFB, but definitely doesn't deduplicate blobs if they are sent in different PFBs. In the test case we send both blobs in a single PFB

to ensure that the second blob will be accepted if I understand @Wondertan correctly.

Reply:
It does have to be a new test case and could be an additional assert on the existing test. The purpose is to make sure that we really have two equal blobs, because existing asserts(on the moment of original comment) could succeed even if there is only one blob.

Reply:
reworked

Reply:
I'm still very confused as to what this is testing in particular. Can you add some docs or further explanation for this test case @vgonkivs ?